src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -98,12 +101,19 @@ public class BackendRegistry {
     private final AuditLog auditLog;
     private final ThreadPool threadPool;
     private final UserInjector userInjector;
-    private final int ttlInMin;
+    private int ttlInMin;
     private Cache<AuthCredentials, User> userCache; // rest standard
     private Cache<String, User> restImpersonationCache; // used for rest impersonation
     private Cache<User, Set<String>> restRoleCache; //
 
     private void createCaches() {
+        Map<AuthCredentials, User> existingUserCache = new HashMap<>();
+        Map<String, User> existingRestImpersonationCache = new HashMap<>();
+        Map<User, Set<String>> existingRestRoleCache = new HashMap<>();
+        if (userCache != null) {
+            existingUserCache.putAll(userCache.asMap());
```

Comment:
Is it necessary to copy the map here? Can't we just say `existingUserCache = userCache.asMap()`?

Reply:
I had this here since the map copy was before the very first time the userCache was defined so it had to handle null.

This is now removed though in favor of starting fresh on a config change.